pkg/reconciler/namespace/namespace.go

```diff
@@ -268,6 +275,15 @@ func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string
 		ns.Labels = map[string]string{}
 	}
 
+	if ns.Labels[clusterLabel] == "" && !scheduleRequirement.Matches(labels.Set(ns.Labels)) {
+		klog.Infof("Skipping placement for namespace %q|%q", lclusterName, ns.Name)
+		return c.patchStatus(ctx, ns, func(conditionsSetter conditions.Setter) {
+			conditions.MarkFalse(conditionsSetter, NamespaceScheduled, NamespaceReasonUnschedulable,
```

Comment:
Maybe a new status reason is suggested to differentiate 'unable to schedule' from 'scheduling disabled'?

Reply:
I agree, I've pushed 395fc13886175bcda5e5a1be685ba7729fbd5c28 (I'll squash once the review is complete), that adds the the `SchedulingDisabled` condition reason.

It has actually highlighted an issue in the condition update logic, where the condition was not created until the namespace was actually scheduled. It's fixed with 395fc13886175bcda5e5a1be685ba7729fbd5c28.